src/lib.rs

```diff
@@ -61,7 +62,13 @@ impl Build {
         {
             Command::new("gmake")
         } else {
-            Command::new("make")
+            if target.contains("wasm") {
+                let mut cmd = Command::new("wasimake");
+
+                std::mem::replace(cmd.arg("make"), Command::new("wasimake"))
```

Comment:
What is `wasimake` and `wasiconfigure`? I don't think these are provided by OpenSSL and otherwise I don't want to rely on too many external programs. (I've also never heard of these myself)

Reply:
> What is `wasimake` and `wasiconfigure`? I don't think these are provided by OpenSSL and otherwise I don't want to rely on too many external programs. (I've also never heard of these myself)

Yes these commands are included in the `wasienv` toolchain https://docs.wasmer.io/ecosystem/wasienv and right now i only found this way to compile openssl for wasm.

Reply:
Can this patch be written without requiring those tools? Without knowing what they do it seems like they could introduce hidden dependencies we aren't aware of.

Reply:
> Can this patch be written without requiring those tools? Without knowing what they do it seems like they could introduce hidden dependencies we aren't aware of.

I dont think so, but this only requires the user to install the wasienv toolchain.

Reply:
Personally I do not want to land this with a dependency on the wasienv toolchain. I don't know why it's necessary as opposed to the standard wasi-sdk.

Reply:
> Personally I do not want to land this with a dependency on the wasienv toolchain. I don't know why it's necessary as opposed to the standard wasi-sdk.

I tried compiling using the standard wasi-sdk but it didnt work no matter what i tried, this seems to be the proper way to configure the build for wasi as far as i know.